web/api/v1/api_test.go

```diff
@@ -41,6 +41,7 @@ import (
 	"github.com/prometheus/common/promlog"
 	"github.com/prometheus/common/route"
 
+	errPkg "github.com/pkg/errors"
```

Comment:
we import this 89 times in our code base and never need to aliase it. why is it needed here?

Reply:
Because of https://github.com/prometheus/prometheus/blob/2cc32c332d1ddf522ada0abe2691d908e91d5182/web/api/v1/api_test.go#L20

Reply:
aren't they compatible? can't we just use pkg/errors?

Reply:
No without an alias.
```Running tool: /usr/bin/go test -timeout 30s github.com/prometheus/prometheus/web/api/v1 -run ^(TestEndpoints)$

# github.com/prometheus/prometheus/web/api/v1 [github.com/prometheus/prometheus/web/api/v1.test]
/home/boyskila/Desktop/prometheus/web/api/v1/api_test.go:44:2: errors redeclared as imported package name
	/home/boyskila/Desktop/prometheus/web/api/v1/previous declaration at ./api_test.go:20:2
/home/boyskila/Desktop/prometheus/web/api/v1/api_test.go:2213:13: undefined: errPkg
FAIL	github.com/prometheus/prometheus/web/api/v1 [build failed]
FAIL
Error: Tests failed.
```

Reply:
```diff
diff --git a/web/api/v1/api_test.go b/web/api/v1/api_test.go
index 50b0191b5..0d141e817 100644
--- a/web/api/v1/api_test.go
+++ b/web/api/v1/api_test.go
@@ -17,7 +17,6 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
@@ -41,7 +40,7 @@ import (
 	"github.com/prometheus/common/promlog"
 	"github.com/prometheus/common/route"
 
-	errPkg "github.com/pkg/errors"
+	"github.com/pkg/errors"
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/pkg/gate"
 	"github.com/prometheus/prometheus/pkg/labels"
@@ -2209,7 +2208,7 @@ func TestParseTimeParam(t *testing.T) {
 				asTime: time.Time{},
 				asError: func() error {
 					_, err := parseTime("baz")
-					return errPkg.Wrapf(err, "Invalid time value for '%s'", "foo")
+					return errors.Wrapf(err, "Invalid time value for '%s'", "foo")
 				},
 			},
 		},
```

That works for me

Reply:
I think you can just replace the existing `errors` import with the new one, since we only use `errors.New()` from `errors`, and `github.com/pkg/errors` also has a `.New()` function.

Reply:
yes, @roidelapluie you are right. it's working. I wasn't completely sure since both packages differ but if you think is fine I'll replace it.

Reply:
> yes, @roidelapluie you are right. it's working. I wasn't completely sure since both packages differ but if you think is fine I'll replace it.

I think we do this in lots of places already, so I would reuse that pattern.